test/js/bun/util/which.test.ts

```diff
@@ -1,11 +1,18 @@
 import { test, expect } from "bun:test";
 
-import { which } from "bun";
+import { $, which } from "bun";
 import { rmSync, chmodSync, mkdirSync, realpathSync } from "node:fs";
 import { join, basename } from "node:path";
 import { tmpdir } from "node:os";
-import { cpSync, rmdirSync } from "js/node/fs/export-star-from";
-import { isIntelMacOS, isWindows } from "../../../harness";
+import { rmdirSync } from "js/node/fs/export-star-from";
+import { isIntelMacOS, isWindows, tempDirWithFiles } from "../../../harness";
+
+{
+  const delim = isWindows ? ";" : ":";
+  if (`${delim}${process.env.PATH}${delim}`.includes(`${delim}.${delim}`)) {
+    throw new Error("$PATH includes . which will mess up the test");
```

Comment:
Can we move this assert into a test block. Could be done as a function `checkPATH()` inside the `test("Bun.which ...` below. That way it doesn't block up the whole file.

Reply:
this check is supposed to block the entire file